libs/vkd3d/resource.c

```diff
@@ -2329,11 +2355,32 @@ HRESULT d3d12_resource_create_committed(struct d3d12_device *device, const D3D12
         else
             allocate_info.heap_flags |= D3D12_HEAP_FLAG_ALLOW_ONLY_NON_RT_DS_TEXTURES;
 
-        allocate_info.vk_image = object->res.vk_image;
+        if (use_dedicated_allocation)
+        {
+            dedicated_info.sType = VK_STRUCTURE_TYPE_MEMORY_DEDICATED_ALLOCATE_INFO;
+            dedicated_info.pNext = NULL;
+            dedicated_info.image = object->res.vk_image;
+            dedicated_info.buffer = VK_NULL_HANDLE;
+            allocate_info.pNext = &dedicated_info;
+        }
+        else
+        {
+            /* We want to allow suballocations and we need the allocation to
+             * be cleared to zero, which only works if we allow buffers */
+            allocate_info.heap_flags &= ~D3D12_HEAP_FLAG_DENY_BUFFERS;
```

Comment:
For NON_ZERO_HEAP I think we don't need buffer memory at all for suballocation.

Reply:
This requires some more thinking, probably best to leave it as-is for now.